### DISCUSSION TITLE
Save and Then Load

### DISCUSSION THREAD

**Discussion by thingsofleon** [lmcinnes/pynndescent#225] [General]
Sorry if I missed in the docs or examples. 

How would I go about saving off the pynndescent model to disk?
How would I load it from disk and run queries against it later (eval or operations)?

**Comment by Petkomat** [comment]
I simply use pickle from Python's standard library.

```
import pickle
import pynndescent

index = pynndescent.NNDescent(xs, ...)

with open("index.pkl", "wb") as f:
    pickle.dump(index, f)

with open("index.pkl", "rb") as f:
    reloaded_index = pickle.load(f)
```

**Reply by thingsofleon** [reply]
ah ok. maybe it is because we are using https://fdasrsf-python.readthedocs.io/en/latest/ for the metrics....

we are getting:
cannot pickle '_cffibackend.FFI' object. 

thoughts? and thanks so much for replying.

**Reply by Petkomat** [reply]
Hi, 

not everything can be pickled (see [this](https://stackoverflow.com/questions/72727104/why-i-am-getting-typeerror-cannot-pickle-cffi-backend-ffi-object-while-execu) for the same (?) error as above, and [this](https://stackoverflow.com/questions/8804830/python-multiprocessing-picklingerror-cant-pickle-type-function) for a slightly general explanation). However, unless you show the code (a minimal example that reproduces the error), it is hard to help.
